### PR TITLE
Add URL param support to toggle connect UI

### PR DESCRIPTION
**Pull request by joshenlim**
Works on all project routes, and if the connectDialog flag is on

Chucked in a few other small fixes too:
- Fix leaving a team redirects back to the landing page instead of the dashboard
- Fix grammar in [`Infrastructure.constants.ts`](https://github.com/supabase/supabase/pull/31241/commits/cc0332f7691aa9caca020dd88cc0da7081f667f7#diff-bdc7a1c23e72a8d24f645ab5fb4ae39910a501a5aaccd2917d129951fdb5e294)
- Add `autoComplete='off'` for TeamSettings search input + DeleteProjectModal input

### CODE REVIEW

File: apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx [hunk 1]
```diff
@@ -119,12 +122,19 @@ const ProjectLayout = forwardRef<HTMLDivElement, PropsWithChildren<ProjectLayout
...
+    const handleMobileMenu = () => {
+      setIsSheetOpen(true)
+    }
 
     useEffect(() => {
       setIsClient(true)
     }, [])
 
+    useEffect(() => {
+      if (showConnect === 'true' && connectDialogUpdate) setShowConnectDialog(true)
```
Comment (MildTomato) [inline apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx:134]: can this not be in connect component?
why does it need to be in a layout?

also need to clean up, remove the param once it's opened so a param isn't hanging around.

**Review by MildTomato — CHANGES_REQUESTED** [review]
requesting change based on last comment
https://github.com/supabase/supabase/pull/31241/files#r1891297222

File: apps/studio/components/interfaces/Connect/Connect.tsx [hunk 2]
```diff
@@ -147,9 +150,20 @@ const Connect = () => {
     selectedGrandchild,
   })
 
+  useEffect(() => {
+    if (showConnect === 'true') setShowConnectDialog(true)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Connect/Connect.tsx:155]: lol

**Review by MildTomato — APPROVED** [review]
looks good. have not checked preview.

**Review by ivasilov — APPROVED** [review]

**Pull request merged**